### ISSUE TITLE
Tweak chatcommand /deleteblocks to affect blocks above and below

### ISSUE THREAD

**Issue by 0-afflatus** [minetest/minetest#2381]
This is a suggestion only.

After line 420 in builtin/game/chatcommands.lua I added
```Lua
p1 = {x = p1.x, y = p1.y, z = p1.z}
p1.y = p1.y + 48
p2.y = p2.y - 48
```

I think this is a better behaviour, but there is probably a better way of coding it.

**Comment by ShadowNinja** [comment]
Why should `/deleteblocks` delete more than you tell it to delete?!

**Comment by 0-afflatus** [comment]
Because it triggers mapgen, which would normally work over a slightly wider area.
Using the 'here' switch an admin is likely to be standing on the ground (or near enough) - doing it this way will regenerate the immediate underground plus any mountainous overhangs caused by mapgen changes.
It means you don't have to keep checking above and below you.
I'm currently fixing a minor but widespread v7 mapgen problem, largely of my own making. This change means I can probably complete the task in one day rather than two.
Perhaps an additional range parameter which defaults to 0 would be better as you wouldn't necessarily want this if you were just repairing griefing.
This follows on from a conversation with est31 on IRC.

**Comment by paramat** [comment]
I see on IRC you are having problems with mgv7 mapgen, could you explain what's happening? are you sure it's 'largely of your own making'? i'm concerned i have done something wrong.
Apparently the just test server is having problems with old parts of the map being regenerated (which should never happen of course), because mgv7 terrain changed many months ago this means discontinuities, is this what is happening to you?
As far as i know nothing i have done would cause this, it seems to be an obscure bug.

**Comment by 0-afflatus** [comment]
@paramat Yes. It's the mountain code. I upgraded rather fast, having started the game in 0.4.10, not realising the potential problems with v7. It doesn't result in "slightly" different mountains, I'm looking at anomalies of up to 20 blocks over a 200-300 node square area. The mapgen I'm using is based on paragenv7, so it doesn't use any of the new features. I also think further problems have ensued because the default mapgen is being triggered in multiplayer/survival mode, but my custom mapgen works in singleplayer/creative. I haven't fully figured what is happening yet.

**Comment by 0-afflatus** [comment]
I think this map describes the situation reasonably well.
![v7a_fixed_150223](https://cloud.githubusercontent.com/assets/7130479/6350281/4eb26ee6-bc2a-11e4-808b-681826557706.png)
I'll take some further screenshots of the anomalies if that would be of any use.

**Comment by 0-afflatus** [comment]
The right-hand side above has been 'fixed' using /deleteblocks. The laft-hand side shows the initial problem.

**Comment by 0-afflatus** [comment]
Today I have to decide whether to continue this process or regenerate the world entirely and copy the builds over using worldedit. I think both approaches will take a couple of days and the latter will be more fun. It has to be said that the regenerated areas look awesome!

**Comment by 0-afflatus** [comment]
We're now way OT for the stated issue, so please moderate as appropriate. I'll try and catch up with you on IRC.

**Comment by 0-afflatus** [comment]
Back on topic - also a 'no-clobber' option which avoids mapchunks containing protected areas would be very useful. Possibly this should be set true by default.

**Comment by paramat** [comment]
Okay thanks, your discontinuities are probably due to older changes to mgv7, there will be more changes soon as i work on a much improved base terrain generation.

**Comment by SmallJoker** [comment]
How about an additional `radius` parameter in `/deleteblock`?

**Comment by 0-afflatus** [comment]
This issue is fixed by #2393 - closing.
